lib/project_types/extension/tasks/go_create_extension.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+require "shopify_cli"
+require "securerandom"
+
+module Extension
+  module Tasks
+    class GoCreateExtension < ShopifyCli::Task
+
+      SPECIFICATION_TO_RENDERER = {
+        "admin_ui_extension": "@shopify/admin-ui-extensions",
+        "checkout_ui_extension": "@shopify/checkout-ui-extensions",
+        "checkout_post_purchase": "@shopify/checkout-ui-extensions",
+      }
+
+      def call(context:, root_dir:, template:, type:)
+        extension_config = Models::ExtensionConfig.new(
+          renderer: SPECIFICATION_TO_RENDERER[type.downcase()],
+          root_dir: root_dir,
+          template: template,
+          type: type,
+          uuid: self.generate_temporary_uuid,
+        )
+        extension_config
+        # TODO
+        # CLI::Kit::System.capture3("./shopify-extensions", "create", `--type=${extension_config.type} ...`)
```

Comment:
this is temp ^^

I am returning the `extension_config` for now, the thought is that it would be used later in the command to create the extension.

Outstanding questions: @t6d 

- Did we decide where the go binary will live? There was some discussion about including it in the CLI itself, instead of on `user/bin`. 
- Confirm: we are not creating a `shopifile.yml` at this point but passing in data as command line args